### PR TITLE
v0.30.1

### PR DESCRIPTION
**Pull request by notgull**
### Added

- Reexport `raw-window-handle` versions 0.4 and 0.5 as `raw_window_handle_04` and `raw_window_handle_05`.
- Implement `ApplicationHandler` for `&mut` references and heap allocations to something that implements `ApplicationHandler`.

### Fixed

- On macOS, fix panic on exit when dropping windows outside the event loop.
- On macOS, fix window dragging glitches when dragging across a monitor boundary with different scale factor.
- On macOS, fix the range in `Ime::Preedit`.

### CODE REVIEW

File: src/changelog/unreleased.md [hunk 1]
```diff
@@ -12,6 +12,7 @@ on how to add them:
 ### Added
 
 - Add `Window::turbo()`, implemented on X11, Wayland, and Web.
+- Add traits `EventLoopExtWayland` and `EventLoopExtX11`, providing methods `is_wayland` and `is_x11` on `EventLoop`.
```
Comment (madsmtm) [inline src/changelog/unreleased.md:15]: I don't think this goes here, that's the example documentation

**Review by madsmtm — APPROVED** [review]
A bit of a large diff (which is totally my own fault), but looks fine, thanks for doing this!

File: src/changelog/unreleased.md (hunk 1 above)
Comment (notgull) [reply src/changelog/unreleased.md]: Good catch, fixed!

**Comment by kchibisov** [comment]
Want to figure https://github.com/rust-windowing/winit/pull/3722 out and then go back to this one.

**Comment by notgull** [comment]
Looks like #3722 is blocked by debate. As I plan to do a weekly release cadence we can just pick it up in the next release.

**Comment by kchibisov** [comment]
Cherry picked recent macOS queue fixes.

**Review by kchibisov — APPROVED** [review]

**Pull request merged**